### PR TITLE
coreclrgc mono

### PR DESCRIPTION
**Pull request by naricc**
This is a draft of changes to enable use of CoreGC on Mono. It builds, but probably doesn't work yet.

### CODE REVIEW

File: src/coreclr/src/gc/env/gcenv.object.h [hunk 1]
```diff
@@ -1,193 +1,10 @@
...
 #ifndef __GCENV_OBJECT_H__
-#define __GCENV_OBJECT_H__
+#define __GCENV_OJECT_H__
 
-// ARM requires that 64-bit primitive types are aligned at 64-bit boundaries for interlocked-like operations.
-// Additionally the platform ABI requires these types and composite type containing them to be similarly
-// aligned when passed as arguments.
-#ifdef TARGET_ARM
-#define FEATURE_64BIT_ALIGNMENT
+#ifdef HAVE_CORE_GC
```
Comment (jkotas) [inline src/coreclr/src/gc/env/gcenv.object.h:4]: This should not be needed here. You should be able to provide Mono-specific `gcenv.h` that has the right environment definition for Mono.

This file is not actually used by the regular CoreCLR GC build. The regular CoreCLR build has its own definition of the environment in https://github.com/dotnet/runtime/blob/master/src/coreclr/src/vm/gcenv.h.

Comment (naricc) [reply src/coreclr/src/gc/env/gcenv.object.h:4]: Thanks. I am trying to work out these details.

**Comment by CoffeeFlux** [comment]
Please let me know when you've responded to Jan's feedback and rebased this to fix the conflicts and I'll take a look.

**Comment by ViktorHofer** [comment]
// Auto-generated message

https://github.com/dotnet/runtime/commit/69e114c1abf91241a0eeecf1ecceab4711b8aa62 which was merged 12/7 removed the intermediate src/coreclr/**src**/ folder. This PR needs to be updated as it touches files in that directory which causes conflicts.

To update your commits you can use this bash script: https://gist.github.com/ViktorHofer/6d24f62abdcddb518b4966ead5ef3783. Feel free to use the comment section of the gist to improve the script for others.

**Comment by naricc** [comment]
We are not moving forward with this for .Net 6.0 for now, so I am closing this PR.

I will continue to keep the branch updated though, and plan to come back to this eventually.

I am also mainting two docs.

The planning doc:https://www.office.com/launch/word/content?auth=2&drive=b!qeja74vwdEu1x4P8cnUQGRTaktYYRzJIko-GJx67iRnB_Z761xo1TIFbsADV8lM-&item=01A27AVOLVFVEKIO34P5F2PXFOBWHNBHWJ&file=https:%2F%2Fmicrosoft-my.sharepoint.com%2Fpersonal%2Fnaricc_microsoft_com%2FDocuments%2FCoreGC%2520on%2520Mono.docx%3Fweb%3D1

And a document describing the changes on a more technical level, as the exist now: https://gist.github.com/naricc/ac858e702633ef60c301c2bec89d7dcc

**Pull request closed without merging**